src/services/cacheService.ts

```diff
@@ -105,17 +115,27 @@ class CacheService {
         config,
         contentType
       )
-      .then(response => response.text())
-      .then(text =>
-        text == ''
-          ? <ActionResponse>{
-              message: 'Cache ' + cacheName + ' created with success',
-              success: true
-            }
-          : <ActionResponse>{ message: text, success: false }
-      )
-      .catch(
-        error => <ActionResponse>{ message: error.toString(), success: false }
+      .then(response => {
+        if (response.ok) {
+          return response.text();
+        }
+        throw response;
+      })
+      .then(text => {
+        let sucessMsg =
+          text == '' ? 'Cache ' + cacheName + ' created with success' : text;
+        return <ActionResponse>{
+          message: sucessMsg,
+          success: true
+        };
+      })
+      .catch(err =>
+        err
+          .text()
+          .then(
+            errorMessage =>
+              <ActionResponse>{ message: errorMessage, success: false }
+          )
```

Comment:
Small thing, but the response handling in `createCacheByConfigName` and `createCacheWithConfiguration` appears to be identical. Can they be combined into a single method that takes a `Promise<Response>` as a param?